karibu-dsl-v23/src/main/kotlin/com/github/mvysny/karibudsl/v23/Dialogs.kt

```diff
@@ -97,6 +116,23 @@ public fun ConfirmDialog.setConfirmIsDanger() {
     setConfirmButtonVariant(ButtonVariant.LUMO_ERROR, ButtonVariant.LUMO_PRIMARY)
 }
 
+/**
+ * Sets the confirm button as dangerous, e.g. when you're confirming a deletion action
+ * that can not be reversed.
+ *
+ * Example of usage:
+ * ```kotlin
+ * setConfirmButton(buildSingleComponent { button("Delete", VaadinIcon.CHECK.create()) {
```

Comment:
This example is very complicated and I'd say it's a misuse of DSL. Far simpler code is produced by simply calling the `Button()` constructor.

Reply:
Fixed